application-task-default/src/main/java/com/xwiki/task/internal/notifications/TaskChangedEventPageWatcher.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package com.xwiki.task.internal.notifications;
+
+import java.util.Arrays;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.filters.watch.WatchedEntitiesManager;
+import org.xwiki.notifications.filters.watch.WatchedEntityFactory;
+import org.xwiki.notifications.filters.watch.WatchedLocationReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+
+import com.xwiki.task.model.Task;
+
+/**
+ * Class which handles auto-watching tasks for users who are assigned to it.
+ *
+ * @version $Id$
+ * @since 3.7
+ */
+@Component
+@Singleton
+@Named("com.xwiki.task.internal.notifications.TaskChangedEventPageWatcher")
+public class TaskChangedEventPageWatcher extends AbstractEventListener
```

Comment:
```suggestion
public class TaskChangedEventListener extends AbstractEventListener
```

This is more suggestive and can easily be reused, in case other actions should be done on `TaskChangedEvent`